ml-algorithms/src/main/java/org/opensearch/ml/engine/utils/TribuoUtil.java

```diff
@@ -47,6 +49,44 @@ public static Tuple<String[], double[][]> transformDataFrame(DataFrame dataFrame
         return new Tuple<>(featureNames, featureValues);
     }
 
+    public static Tuple<String[], double[][]> transformClassificationDataFrame(DataFrame dataFrame, String target) {
+        List<String> featureNames = Arrays.stream(dataFrame.columnMetas()).map(ColumnMeta::getName).collect(Collectors.toList());
+        int targetIndex = findFinalTargetIndex(featureNames, target);
+
+        int i = 0;
+        Iterator<Row> itr = dataFrame.iterator();
+        double[][] featureValues = new double[dataFrame.size()][featureNames.size() - 1];
+        while (itr.hasNext()) {
+            Row row = itr.next();
+            int col = 0;
+            for (int j = 0; j < featureNames.size(); j++) {
+                if (j == targetIndex) {
+                    continue;
+                }
+                featureValues[i][col++] = row.getValue(j).doubleValue();
+            }
+            ++i;
+        }
+        featureNames.remove(target);
+        return new Tuple<>(featureNames.toArray(new String[featureNames.size()]), featureValues);
+    }
+
+    public static String[] transformTargetValuesDataFrames(DataFrame dataFrame, String target) {
+        List<String> featureNames = Arrays.stream(dataFrame.columnMetas()).map(ColumnMeta::getName).collect(Collectors.toList());
+        int targetIndex = findFinalTargetIndex(featureNames, target);
```

Comment:
Line 75-67 seems duplicate code with line 53-54, maybe add one method `int getColumnIndex(String targetColumn)` in DataFrame ?

Reply:
Yes it's doable.